app/src/main/java/com/quran/labs/androidquran/ui/util/ToastCompat.java

```diff
@@ -0,0 +1,83 @@
+package com.quran.labs.androidquran.ui.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
+import androidx.appcompat.content.res.AppCompatResources;
+import androidx.core.content.ContextCompat;
+import androidx.core.graphics.drawable.DrawableCompat;
+
+import com.quran.labs.androidquran.R;
+import com.quran.labs.androidquran.util.QuranSettings;
+
+public class ToastCompat {
+
+  private static final int ANDROID_R = 30;
+
+  public static Toast makeText(@NonNull Context context, @StringRes int messageRes, int duration) {
+    return makeText(context, context.getString(messageRes), duration);
+  }
+
+  public static Toast makeText(@NonNull Context context, CharSequence message, int duration) {
+    @SuppressLint("ShowToast") Toast toast = Toast.makeText(context, message, duration);
+    if (Build.VERSION.SDK_INT < ANDROID_R) {
+      // It's unclear what behavior `Toast.setView()` has on R and above, hence the guard.
+      toast.setView(createThemeFriendlyToastView(context, message));
```

Comment:
I suspect that this would still continue to work on Android R (`Toast.getView()`/`Toast.setView(View)` is deprecated from that version and I read that `getView()` will return `null`) but I haven't been able to test it. 

When it can be tested/when the behaviour will be clearer, this guard could be removed/amended.